app/assets/javascripts/discourse/app/controllers/user-private-messages.js

```diff
@@ -2,11 +2,27 @@ import Controller, { inject as controller } from "@ember/controller";
 import { action, computed } from "@ember/object";
 import { inject as service } from "@ember/service";
 import { alias, and, equal, readOnly } from "@ember/object/computed";
-import { tracked } from "@glimmer/tracking";
+import { cached, tracked } from "@glimmer/tracking";
 import I18n from "I18n";
 import DiscourseURL from "discourse/lib/url";
 
-export const PERSONAL_INBOX = "__personal_inbox__";
+const customUserNavMessagesDropdownRows = [];
+
+export function registerCustomUserNavMessagesDropdownRow(
+  routeName,
+  name,
+  icon
+) {
+  customUserNavMessagesDropdownRows.push({
+    routeName,
+    name,
+    icon,
+  });
+}
+
+export function resetCustomUserNavMessagesDropdownRows() {
+  customUserNavMessagesDropdownRows.length = 0;
```

Comment:
This is interesting way to reset. Is there a benefit of this approach over `= []`?

Reply:
Setting `.length = 0` clears the original array whereas `= []` is setting the variable to another array. If there is a spot in the code that has already referenced this constant, using `= []` will actually not work properly.